api/NESSIE-SPEC-2-0.md

```diff
@@ -0,0 +1,45 @@
+# Nessie Specification
+
+Nessie Specifications define the behaviour of Nessie Servers. The same behaviour can be expected to be observable
+via all supported API (unless noted otherwise).
+
+[Semver](https://semver.org/spec/v2.0.0.html) principles are followed in bumping the Nessie spec version.
+For example, a major behaviour change will result in incrementing the major version number.
+
+The patch version number will be incremented only on minute spec corrections that do not significantly alter the 
+server's behaviour.
+
+If a server returns a specific spec version via `NessieConfiguration.getSpecVersion()` it MUST comply with the
+specified behaviour (see sections below).
+
+Servers SHOULD return the spec version that defines its behaviour most completely.
+
+A server SHOULD return `null` from `NessieConfiguration.getSpecVersion()`, when it behaves in a way that is not
```

Comment:
Servers don't reply with this field, if the client did not send the corresponding header (so that the server assumes that the client cannot handle the new attributes in NC).

Reply:
Not quite... `getSpecVersion()` is only affected by the API version, not by headers, AFAIK. The API version restriction should hopefully be clear to users from java annotations and OpenAPI tags :thinking: